### ISSUE TITLE
Occasional "invalid character 'R' looking for beginning of value" errors

### ISSUE THREAD

**Issue by DuaneOBrien** [depscloud/hacktoberfest#5]
While running the `identify-contribution-candidates` utility against our internal depscloud instance, I'm occasionally seeing "Invalid character 'R'" errors, see below: 

```
2020/10/14 17:01:58 invalid character 'R' looking for beginning of value
2020/10/14 17:01:58 filtering company module language:"java" organization:"indeed" module:(redacted) 
2020/10/14 17:01:58 filtering company module language:"java" organization:"com.indeed" module:(redacted) 
2020/10/14 17:01:59 invalid character 'R' looking for beginning of value
2020/10/14 17:01:59 filtering company module language:"go" organization:"indeed.com" module:(redacted)
2020/10/14 17:01:59 invalid character 'R' looking for beginning of value
2020/10/14 17:01:59 filtering company module language:"node" organization:"indeed" module:(redacted)
2020/10/14 17:01:59 invalid character 'R' looking for beginning of value
```

Do you have any sense for what would be the source of the errors?

**Comment by mjpitz** [comment]
I vaguely recall hitting his issue with API calls to libraries.io. Do you have the API key set up? I can definitely look into adding some more logging to help troubleshoot this more. If I recall, I kinda threw this together in like a night or something like that.

**Comment by DuaneOBrien** [comment]
I do have the API Key set up yeah. It ran for about 90 minutes and right now it's just hanging. I don't know if it's compiling the candidates.json file or just a network issue. I'm gonna walk away and see if it resolves.

**Comment by mjpitz** [comment]
Last time I ran it, it was running for close to 3/4 hours. It usually looks like it gets stuck when it's compiling a big tree because there's no progress output. But yes. Watch your VPN. That bit me.

**Comment by DuaneOBrien** [comment]
It finally un-hung about 8 hours later. Might have been a VPN issue. It threw this:

`2020/10/14 23:39:42 rpc error: code = Unavailable desc = transport is closing`

**Comment by mjpitz** [comment]
it was probably the VPN. The process is not super great. There's a lot of caching that can be done and a lot of parallelization too.

**Comment by DuaneOBrien** [comment]
I mostly tripped over all of this while I was working to generate a candidate.json file for this issue:

https://github.com/indeedeng/Mariner-Issue-Collector/issues/11

**Comment by mjpitz** [comment]
For what it's worth @DuaneOBrien, it does try and fail gracefully and output what information it does not to a file. Not great because it's not every module like you would expect, but it should still give you a rough idea. 

Unfortunately, I wasn't able to get to this last night as I was trying to wrap up the last stepping stones toward `v1beta`. I do have a good bit of time this weekend and should be able to get some work in on this.

**Comment by mjpitz** [comment]
Also, I'd love your help trying to figure out the right way to handle a use case. Right now, when a module resolves to the same source, we sum their scores. `babel` and many  larger NodeJS projects use lerna.js to manage a monorepo. These types of monorepos produce many NodeJS modules (vs how `depscloud` does it where it's a single module with many artifacts). As a result, this can lead to an inflation of the scores for a given source. Any thoughts on how you'd score this kind of thing?

**Comment by DuaneOBrien** [comment]
No worries on not getting to it, it'll come when it comes. 

I did see that it output the data it had, which was really useful. It got me where I needed to go for Mariner.

**Comment by DuaneOBrien** [comment]
Re: Score Inflation

For me, the intent behind the score is to provide a first order of approximation for how important it is to pay attention to any given dependency or repository. In that context, I think the score inflation is probably fine. We already know it's going to be imprecise on some level. For my use case, it's a wash - I just want a list of things that we use so I can go look for issues and put those issues in an approximate order of "how important is this dependency to us." But if folks start using the scores differently, and the difference between 10th and 12th place on the list means something, then it will probably start to matter a lot.

**Comment by mjpitz** [comment]
Ok. So I think I resolved this issue. I don't think libraries.io implemented their rate-limiting properly. I am literally sending 1 request / second and saw I was still being rate-limited on a fraction of my requests.. For the most part, this has been alleviated. In the recent release, I also changed how the computation was being done so it should be much faster than before. Let me know if the 'R' parsing is a rampant issue (it'll probably still happen, I just tried to make it happen less frequently).
